### PR TITLE
hypershift creds mount optional

### PR DESCRIPTION
**Pull request by SANJEEV-Choubey**
Remove mount and secret from hypershift deployment, when not provided.
https://github.com/openshift/hypershift/issues/651

### CODE REVIEW

**Comment by relyt0925** [comment]
@SANJEEV-Choubey Could we also disable the deployment of the following secrets when not necessary
```
+  - apiVersion: v1
+    data:
+      credentials: null
+    kind: Secret
+    metadata:
+      creationTimestamp: null
+      name: hypershift-operator-aws-credentials
+      namespace: hypershift
+  - apiVersion: v1
+    data:
+      credentials: null
+    kind: Secret
+    metadata:
+      creationTimestamp: null
+      name: hypershift-operator-oidc-provider-s3-credentials
+      namespace: hypershift
+  - apiVersion: v1
+    data:
+      name: ""
+      region: ""
+    kind: ConfigMap
+    metadata:
+      creationTimestamp: null
+      name: oidc-storage-provider-s3-config
+      namespace: kube-public
```

File: cmd/install/install.go [hunk 1]
```diff
@@ -273,15 +273,16 @@ func hyperShiftOperatorManifests(opts Options) ([]crclient.Object, error) {
...
 	objects = append(objects, operatorService)
 	objects = append(objects, prometheusRole)
 	objects = append(objects, prometheusRoleBinding)
 	objects = append(objects, serviceMonitor)
 	objects = append(objects, recordingRule)
-	objects = append(objects, assets.OIDCStorageProviderS3ConfigMap(opts.OIDCStorageProviderS3BucketName, opts.OIDCStorageProviderS3Region))
-
+	if opts.OIDCStorageProviderS3BucketName != "" {
+		objects = append(objects, operatorOIDCProviderS3Secret)
+		objects = append(objects, operatorCredentialsSecret)
```
Comment (sjenning) [inline cmd/install/install.go:284]: This secret is related to private cluster operations on AWS, not OIDC storage.  Thus it should not be tied into the bucket name being specified.

**Review by sjenning — CHANGES_REQUESTED** [review]

**Pull request merged**